evm/src/executor/mod.rs

```diff
@@ -219,7 +219,32 @@ impl Executor {
         // and also the chainid, which can be set manually
         self.env.cfg.chain_id = res.env.cfg.chain_id;
 
-        Ok(res)
+        match res.state_changeset.as_ref() {
+            Some(changeset) => {
+                let success = self
+                    .ensure_success(to, res.reverted, changeset.clone(), false)
+                    .map_err(|err| EvmError::Eyre(eyre::eyre!(err.to_string())))?;
+                if success {
+                    Ok(res)
+                } else {
+                    Err(EvmError::Execution {
+                        reverted: res.reverted,
+                        reason: "execution error".to_owned(),
+                        traces: res.traces.clone(),
+                        gas_used: res.gas_used,
+                        gas_refunded: res.gas_refunded,
+                        stipend: res.stipend,
+                        logs: res.logs.clone(),
+                        debug: res.debug.clone(),
+                        labels: res.labels.clone(),
+                        state_changeset: None,
+                        transactions: None,
+                        script_wallets: res.script_wallets.clone(),
```

Comment:
why do we need to clone everything here?

can't we just return here?

Reply:
yup, sry, was debugging smth and forgot to clean up